src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -398,3 +398,99 @@ ScanStateGetExecutorState(CitusScanState *scanState)
 {
 	return scanState->customScanState.ss.ps.state;
 }
+
+
+/*
+ * FetchCitusCustomScanIfExists traverses a given plan and returns a Citus CustomScan
+ * if it has any.
+ */
+CustomScan *
+FetchCitusCustomScanIfExists(Plan *plan)
+{
+	CustomScan *customScan = NULL;
+
+	if (plan == NULL)
+	{
+		return NULL;
+	}
+
+	if (IsCitusCustomScan(plan))
+	{
+		return (CustomScan *) plan;
+	}
+
+	customScan = FetchCitusCustomScanIfExists(plan->lefttree);
+
+	if (customScan == NULL)
+	{
+		customScan = FetchCitusCustomScanIfExists(plan->righttree);
+	}
+
+	return customScan;
+}
+
+
+/*
+ * IsCitusPlan returns whether a Plan contains a CustomScan generated by Citus
+ * by recursively walking through the plan tree.
+ */
+bool
+IsCitusPlan(Plan *plan)
+{
+	if (plan == NULL)
+	{
+		return false;
+	}
+
+	if (IsCitusCustomScan(plan))
+	{
+		return true;
+	}
+
+	if (plan->lefttree != NULL && IsCitusPlan(plan->lefttree))
```

Comment:
At the beginning of the function you have the null check, so you instead of the following part you can just return this:

```
return IsCitusPlan(plan->leftTree) || isCitusPlan(plan->rightTree)
```

Reply:
makes sense. thanks for the advice